packages/x-data-grid/src/material/index.tsx

```diff
@@ -75,6 +75,37 @@ const InputAdornment = styled(MUIInputAdornment)({
   },
 });
 
+const FormControlLabel = styled(MUIFormControlLabel, {
+  shouldForwardProp: (prop) => prop !== 'fullWidth',
+})<{ fullWidth?: boolean }>(({ theme }) => ({
+  gap: theme.spacing(0.5),
+  margin: 0,
```

Comment:
Is this something completely material specific, or could it be styled in the core grid?

Any styling in the material bindings is not going to be available for other design-systems, so I'd try to keep as much styles as possible on the core.